lib/os/heap.c

```diff
@@ -290,9 +329,40 @@ void *sys_heap_aligned_alloc(struct sys_heap *heap, size_t align, size_t bytes)
 	}
 
 	set_chunk_used(h, c, true);
+
+	if (c0 == h->top_available) {
+		update_top_available(h);
+	}
+
+	sys_heap_print_stats(heap);
+
 	return mem;
 }
 
+void heap_print_stats(struct z_heap *h)
```

Comment:
This probably belongs in heap-validate.c and not heap.c.  Also might consider using the logging subsystem instead of printk(), since this is higher level stuff.